frontend/src/store/modules/auth.ts

```diff
@@ -41,6 +43,32 @@ const getters = {
 };
 
 const actions = {
+  async exchangeOAuthToken(
```

Comment:
Let's move this to a separate oauth module. And let's define a new API endpoint POST /api/oauth/vcs/{vcsID}/exchange-token

The reason is 

1. oauth is a different thing from auth
2. We may have different oauth provider later (vcs is only one of them)
3. exchange token is considered a POST instead of GET since it causes side-effect (generates a new token).

Reply:
Needs to change frontend as well. We need to put it under oauth.ts